third_party/terraform/utils/error_retry_predicates.go

```diff
@@ -1,23 +1,82 @@
 package google
 
 import (
-	"google.golang.org/api/googleapi"
+	"fmt"
 	"log"
+	"net/url"
 	"strings"
+
+	"google.golang.org/api/googleapi"
 )
 
+type RetryErrorPredicateFunc func(error) (bool, string)
+
+var defaultErrorRetryPredicates = []RetryErrorPredicateFunc{
+	isUrlTimeoutError,
+	isOperationInProgressError,
+	isCommonRetryableErrorCode,
+}
+
+func isUrlTimeoutError(err error) (bool, string) {
+	if urlerr, ok := err.(*url.Error); ok && urlerr.Timeout() {
+		log.Printf("[DEBUG] Dismissed an error as retryable based on googleapis.com target: %s", err)
+		return true, "Got URL timeout error"
+	}
+	return false, ""
+}
+
+func isOperationInProgressError(err error) (bool, string) {
+	gerr, ok := err.(*googleapi.Error)
+	if !ok {
+		return false, ""
+	}
+
+	if gerr.Code == 409 && strings.Contains(gerr.Body, "operationInProgress") {
+		// 409's are retried because some APIs (cloud sql) throws a 409 when concurrent
```

Comment:
I'd love for comments like this one to live as the function comments now, and for a brief plain-english comment breaking down what our common retries are retrying for. Not needed, just nice-to-have.